pkg/types/chain_reader.go

```diff
@@ -0,0 +1,76 @@
+package types
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/grpc/status"
+)
+
+// Errors exposed to product plugins
+
+type errChainReader string
+
+func (e errChainReader) Error() string { return string(e) }
+
+const (
+	ErrInvalidType   = errChainReader("invalid type")
+	ErrFieldNotFound = errChainReader("field not found")
+)
+
+func UnwrapClientError(err error) error {
+	if s, ok := status.FromError(err); ok {
+		return errChainReader(s.String())
+	}
+	return err
+}
+
+// Errors used only by relay plugins
+
+type ErrorChainReaderUnsupported struct{}
+
+func (e ErrorChainReaderUnsupported) Error() string {
+	return "ChainReader is not supported by the relay"
+}
+
+type ErrorNoChainReaderInJobSpec struct{}
```

Comment:
Why does this need to be available for the chain relayer repos? They don't parse their own job specs - that is a core concept only.

Reply:
It's a part of the `RelayConfig`. My impression was that it was the relay's job to parse `RelayConfig`, it's supposed to be opaque to core isn't it?

Reply:
I think I see your point though--I guess we should rename the error to `ErrorNoChainReaderInRelayConfig`?

Reply:
The error message should mention something about the job spec, so the user realizes that's where the problem is... but I guess we can wrap this one with a more general message from core about the job spec being the problem

Reply:
The relay produces the error, core consumes there error

Reply:
(Well, to be more precise: in what's checked in the relay produces the error and consumes the error, but on my local branch I'm changing it to go over the wire so that core will be able to consume it)